From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Tue, 7 May 2013 12:57:42 -0700
Subject: [PATCH] Fix 32bit kernel builds by not using eu-strip

Right now 32bit kernel builds are failing because eu-strip is not
performing safe operations. In order to work around this, use objdump
as before when running a kernel build (by checking for a Kconfig file
in the BUILDIR).

[2021-09-10]: Ported for migration to debugedit 5.0 when updating to rpm
4.17.0

Change-Id: I8c24eaab9e0ac1dfe21484522ce4e415d21fcb14
Ported-by: Patrick McCarty <patrick.mccarty@intel.com>
Signed-off-by: William Douglas <william.douglas@intel.com>
Signed-off-by: Patrick McCarty <patrick.mccarty@intel.com>
---
 scripts/find-debuginfo.in | 34 +++++++++++++++++++++++++++++-----
 1 file changed, 29 insertions(+), 5 deletions(-)

diff --git a/scripts/find-debuginfo.in b/scripts/find-debuginfo.in
index 8b9ce77..7be7ad5 100755
--- a/scripts/find-debuginfo.in
+++ b/scripts/find-debuginfo.in
@@ -481,12 +481,36 @@ do_file()
   esac
 
   mkdir -p "${debugdn}"
-  if test -w "$f"; then
-    strip_to_debug "${debugfn}" "$f"
+  if [ -e "${BUILDDIR}/Kconfig" ] ; then
+      mode=$(stat -c %a "$f")
+      chmod +w "$f"
+      objcopy --only-keep-debug $f $debugfn || :
+      (
+	  shopt -s extglob
+	  strip_option="--strip-all"
+	  case "$f" in
+	      *.ko)
+		  strip_option="--strip-debug" ;;
+	      *$STRIP_KEEP_SYMTAB*)
+		  if test -n "$STRIP_KEEP_SYMTAB"; then
+		      strip_option="--strip-debug"
+		  fi
+		  ;;
+	  esac
+	  if test "$NO_DEBUGINFO_STRIP_DEBUG" = true ; then
+	      strip_option=
+	  fi
+	  objcopy --add-gnu-debuglink=$debugfn -R .comment -R .GCC.command.line $strip_option $f
+	  chmod $mode $f
+      ) || :
   else
-    chmod u+w "$f"
-    strip_to_debug "${debugfn}" "$f"
-    chmod u-w "$f"
+      if test -w "$f"; then
+	  strip_to_debug "${debugfn}" "$f"
+      else
+	  chmod u+w "$f"
+	  strip_to_debug "${debugfn}" "$f"
+	  chmod u-w "$f"
+      fi
   fi
 
   # strip -g implies we have full symtab, don't add mini symtab in that case.
